spec/hashcash_spec.cr

```diff
@@ -1,161 +1,72 @@
 require "./spec_helper"
 
 describe Hashcash do
-  it "should initialise a new hashcash instance" do
-    # with just resource arg
-    new_stamp = Hashcash::Stamp.new("gab@place.technology")
-
-    new_stamp.stamp_string.should eq ""
-    new_stamp.resource.should eq "gab@place.technology"
-    new_stamp.bits.should eq 20
-    new_stamp.date.hour.should eq Time.utc.hour
-    new_stamp.version.should eq 1
-
-    # with all of the args
-    custom_stamp = Hashcash::Stamp.new("hi@hello.com", 2, 16, Time.utc, "goodbye")
-
-    custom_stamp.stamp_string.should eq ""
-    custom_stamp.resource.should eq "hi@hello.com"
-    custom_stamp.version.should eq 2
-    custom_stamp.bits.should eq 16
-    custom_stamp.date.hour.should eq Time.utc.hour
-  end
-
-  # test generate class method
-  it "should generate a hashcash stamp string" do
-    new_stamp = Hashcash::Stamp.new("hello")
-    new_stamp_string = new_stamp.generate
-
-    new_stamp_string.should contain "hello"
-    new_stamp_string[0].should eq '1'
-    new_stamp_string.should contain "1:20:"
-    new_stamp.stamp_string.should contain "hello"
-
-    # with all of the args
-    custom_stamp = Hashcash::Stamp.new("hi@hello.com", 2, 16, Time.utc(2016, 2, 15, 10, 20, 30), "goodbye")
-    custom_stamp_string = custom_stamp.generate
-
-    custom_stamp_string.should contain "2:16:160215102030:hi@hello.com:goodbye:"
-  end
-
   # test Hashcash.generate
   it "should initilaise and generate a hashcash stamp using the high level generate method" do
```

Comment:
For naming and structuring specs, there are some excellent guidelines at https://www.betterspecs.org/.